common.go

```diff
@@ -32,45 +36,45 @@ type UserStore interface {
 // map[service]map[method]T
 type Config[T any] map[string]map[string]T
 
-// returns the config value for the given request.
-func (c Config[T]) Get(r *rcpRequest) (v T, ok bool) {
+// Get returns the config value for the given request.
+func (c Config[T]) Get(r *Request) (v T, ok bool) {
 	if c == nil {
 		return v, false
 	}
 
-	methodCfg, ok := c[r.serviceName][r.methodName]
+	methodCfg, ok := c[r.ServiceName][r.MethodName]
 	if !ok {
 		return v, false
 	}
 
 	return methodCfg, true
 }
 
-// rcpRequest is a parsed RPC request.
-type rcpRequest struct {
-	packageName string
-	serviceName string
-	methodName  string
+// Request is a parsed RPC request.
+type Request struct {
+	PackageName string
+	ServiceName string
+	MethodName  string
 }
 
 // newRequest parses a path into an rcpRequest.
-func newRequest(path string) *rcpRequest {
+func ParseRequest(path string) *Request {
```

Comment:
Lets not export this function if we need it somwhere else, i would copy it, to keep the API of this package simple.

Reply:
I would rather keep it exported since I'm using it here:
https://github.com/0xsequence/quotacontrol/pull/53/files#diff-7bf08ce0a790451f4677b47d7e17aafcc1825aa92312c83775de6c22e7b5a102R10-R23

Reply:
This would panic if the path is not valid webrpc route. Because you don't check for nil. And we don't check for nil in `Get()` function.
This is not good to export these functions, to be able to just pass the result without checking to another function.

So, i suggest to merge the two functions together. And get rid of the whole `webrpcRequest` struct, it is not needed after all.

This is smaller and more understandable code.

```go
// Config is a generic map of services/methods to a config value.
// map[service]map[method]T
type Config[T any] map[string]map[string]T

// Get returns the config value for the given request.
func (c Config[T]) Get(path string) (*T, error) {
	if c == nil {
		return nil, fmt.Errorf("cofig is nil")
	}

	p := strings.Split(path, "/")
	if len(p) < 4 {
		return nil, fmt.Errorf("path has not enough parts")
	}

	var (
		packageName = p[len(p)-3]
		serviceName = p[len(p)-2]
		methodName  = p[len(p)-1]
	)

	if packageName != "rpc" {
		return nil, fmt.Errorf("path doesn't include rpc")
	}

	methodCfg, ok := c[serviceName][methodName]
	if !ok {
		return nil, fmt.Errorf("acl not found")
	}

	return &methodCfg, nil
}
```

And here: https://github.com/0xsequence/authcontrol/blob/master/middleware.go#L135-L145
Replace it with just:
```go
acl, err := acl.Get(r.URL.Path)
if err != nil {
	eh(r, w, proto.ErrUnauthorized.WithCausef("get acl: %w", err))
	return
}
```